include/xio_user.h

```diff
@@ -483,11 +510,26 @@ uint32_t xio_lookup_rkey_by_response(const struct xio_reg_mem *reg_mem,
  *
  * @return 0 on success, or -1 on error.  If an error occurs, call
  *	    xio_errno function to get the failure reason.
+ * @deprecated use @ref xio_mem_alloc_ex
  */
 int xio_mem_alloc(size_t length, struct xio_reg_mem *reg_mem);
 
 /**
- * free registered memory region, create by @ref xio_mem_alloc
+ * allocate and register memory
+ *
+ * @param[in] length	length of required buffer memory.
+ * @param[out] reg_mem	registered memory data structure
+ * @param[in] opt	options that determine if the memory will be registered.
+ *
+ * @return 0 on success, or -1 on error.  If an error occurs, call
+ *	    xio_errno function to get the failure reason.
+ */
+int xio_mem_alloc_ex(size_t length, struct xio_reg_mem *reg_mem,
```

Comment:
i didn't do it because:

1. length is a mandatory while xio_mem_alloc_params is not. ( i though of adding support that it can be null.
1. i wanted to save a creation of another struct, i'm also using struct xio_mem_alloc_params in the set_opt, and length should not be in the set_opt
but i don't mind changing it